docs/sections/life_of_a_workflow.md

```diff
@@ -0,0 +1,111 @@
+# Life of a Workflow
+
+This section explains what popper does when it executes a workflow. We will break down what popper does behind the scenes when executing the sample workflow described [here](https://github.com/getpopper/popper/blob/master/docs/sections/getting_started.md).
+
+Each step of a workflow has the following stages:
+
+## 1. Look at `uses` attribute and pull/build image
+
+Each step of a workflow must specify the `DockerFile` images it requires with a `uses` line. For example, the first step of our example workflow contains the following line:
```

Comment:
We cannot reference an image as "a `Dockerfile` image", rather an image is built from a `Dockerfile`, and a container is instantiated from an image. The `uses` keyword references either a `Dockerfile` or an image. If it is referencing a `Dockerfile`, Popper builds a container image first, then it instantiates a container from it.

Reply:
Ah, thank you for the explanation! We have reworded that line to be more accurate.